### ISSUE TITLE
"PKIX path building failed" when connecting to Microsoft SQL Server

### ISSUE THREAD

**Issue by eirikbakke** [ultorg/public_issues#2]
When connecting Ultorg to a Microsoft SQL Server instance, some users reported an error in the New Connection Wizard starting "PKIX path building failed".

One workaround is to add ";trustServerCertificate=true" to the end of the JDBC URL.

The connection will still be encrypted, but the client won't try to verify the identity of the server. Alternatively, the latter security feature can be configured with the trustStore property.

More information on these settings here:
https://docs.microsoft.com/en-us/sql/connect/jdbc/connecting-with-ssl-encryption?view=sql-server-ver15
https://docs.microsoft.com/en-us/sql/connect/jdbc/understanding-ssl-support?view=sql-server-ver15
https://docs.microsoft.com/en-us/sql/connect/jdbc/configuring-the-client-for-ssl-encryption?view=sql-server-ver15

The TODO for Ultorg will probably be to document these properties within the connection UI. Or perhaps the "-Djavax.net.ssl.trustStoreType=WINDOWS-ROOT" property should be set at the application level; I noticed that DBeaver does this by default.

**Comment by davrax** [comment]
Ran into this error today- the workaround unblocked me. One thing I would've looked for is the ability to configure SQL Server Auth in the Ultorg Connection UI- specifying Windows Auth vs SQL Server Auth, SSL Enabled Y/N, etc. 

I'm one of (presumably few) users working from an Apple Silicon Mac and MS SQL Server, so the ODBC/JDBC driver situation alone is _complex_.

**Comment by eirikbakke** [comment]
@davrax Yeah, I'd love to simplify the connection experience. The challenge is, there are so many different combinations of setups, and it would be hard to test them all.

A short-term solution I'm thinking of is to add a sidebar with a FAQ/documentation in the connection dialog itself, e.g. so that workarounds like these could be documented right next to where you need to type them in. And I could add links to the official documentation etc.

By the way, in your case, was there any problems with the bundled JDBC driver?

**Comment by davrax** [comment]
@eirikbakke Understandable- I think that'd be sufficient for your current group of users. No issues with your bundled JDBC driver (the issues I mentioned were with another vendor's software), and thanks for compiling a native ARM64 .dmg!

**Comment by eirikbakke** [comment]
Great! Yeah, the new MacBooks are super fast when you give them the right binaries... I want one now.

**Comment by eirikbakke** [comment]
There's a new documentation site for Ultorg now, which gathers all the information about connecting to Microsoft SQL Server here:
https://www.ultorg.com/docs/connect/mssql/

I have opened a separate issue for improvements to the connection UI that would simplify the process of dealing with certificates:
https://github.com/ultorg/public_issues/issues/79
